src/edu/ucsd/library/xdre/tab/TabularEditRecord.java

```diff
@@ -0,0 +1,223 @@
+package edu.ucsd.library.xdre.tab;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.dom4j.Document;
+import org.dom4j.DocumentException;
+import org.dom4j.Element;
+import org.dom4j.Node;
+
+import edu.ucsd.library.xdre.utils.Constants;
+
+
+/**
+ * Tabular data consisting of a key-value map for the record, and 0 or more
+ * components (also key-value maps) for editing.
+ * @author lsitu
+**/
+public class TabularEditRecord extends TabularRecord
+{
+    public static String[] NONE_DESCRIPTIVE_NODES = {"hasFile", "hasComponent"};
+
+    private Document document = null;
+
+    /**
+     * Create an empty record.
+    **/
+    public TabularEditRecord() {
+        this( null );
+    }
+
+    /**
+     * Create a record with record-level data, but no component-level data.
+    **/
+    public TabularEditRecord(Map<String,String> data) {
+        super( data );
+    }
+
+    /**
+     * Create a record with both record-level and component-level data.
+     * @throws DocumentException 
+     * @throws IOException 
+     * @throws UnsupportedEncodingException 
+    **/
+    public TabularEditRecord( Map<String,String> data, ArrayList<TabularRecord> cmp, Document document )
+            throws UnsupportedEncodingException, IOException, DocumentException
+    {
+        super( data, cmp );
+        editDocument(document);
+    }
+
+    public void editDocument(Document document) throws UnsupportedEncodingException, IOException, DocumentException {
+        this.document = document;
+    }
+
+    /**
+     * Convert the record to RDF/XML.
+     * @throws Exception 
+    **/
+    public Document toRDFXML() throws Exception
+    {
+        return editDocument();
+    }
+
+    /**
+     * Edit the document with the excel data provided 
+     * @return
+     * @throws Exception 
+     */
+    protected Document editDocument() throws Exception {
+        overlayElements(data);
+
+        return document;
+    }
+
+    /**
+     * Convert ark value to ARK URL. 
+     * @param value
+     * @return
+     */
+    public static String getArkUrl(String value) {
+        String arkValue = value;
+
+        if (arkValue.startsWith(Constants.DAMS_ARK_URL_BASE)) {
+            return value;
+        } else {
+            // Convert ARK to dams4 ARK url
+            return toArkUrl(arkValue);
+        }
+    }
+
+    /*
+     * Convert ark to dams4 ARK url
+     * @param ark
+     * @return
+     */
+    private static String toArkUrl(String ark) {
+        if (!(ark.startsWith("http://") || ark.startsWith("https://"))) {
+            String arkUrlBase = Constants.DAMS_ARK_URL_BASE;
+            arkUrlBase += arkUrlBase.endsWith("/") ? "" : "/";
+            return arkUrlBase + "20775/" + ark;
```

Comment:
In the test, you use `Constants.ARK_ORG`, would that work here as well?

Reply:
Yes, I think we should use that `Constants.ARK_ORG` instead. Thanks.